src/Core/Grid/Definition/Factory/LogGridDefinitionFactory.php

```diff
@@ -0,0 +1,186 @@
+    private $templating;
+
+    /**
+     * @param EngineInterface $templating
+     */
+    public function __construct(
+        EngineInterface $templating
+    ) {
+        $this->templating = $templating;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getIdentifier()
+    {
+        return 'logs';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName()
+    {
+        return $this->trans('Logs', [], 'Admin.Advparameters.Feature');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns()
+    {
+        $templating = $this->templating;
+        $displayEmployee = function ($row) use ($templating) {
+            return $templating->render('@AdvancedParameters/LogsPage/Blocks/employee_block.html.twig', [
+                'log' => $row,
+            ]);
+        };
+
+        $columnsArray = [
+            [
+                'identifier' => 'id_log',
+                'name' => $this->trans('ID', [], 'Admin.Global'),
+                'filter_form_type' => TextType::class,
+            ],
+            [
+                'identifier' => 'employee',
+                'name' => $this->trans('Employee', [], 'Admin.Global'),
+                'filter_form_type' => TextType::class,
+                'modifier' => $displayEmployee,
+                'raw_content' => true,
+            ],
+            [
+                'identifier' => 'severity',
+                'name' => $this->trans('Severity (1-4)', [], 'Admin.Advparameters.Feature'),
+                'filter_form_type' => TextType::class,
+            ],
+            [
+                'identifier' => 'message',
+                'name' => $this->trans('Message', [], 'Admin.Global'),
+                'filter_form_type' => TextType::class,
+            ],
+            [
+                'identifier' => 'object_type',
+                'name' => $this->trans('Object type', [], 'Admin.Advparameters.Feature'),
+                'filter_form_type' => TextType::class,
+            ],
+            [
+                'identifier' => 'object_id',
+                'name' => $this->trans('Object ID', [], 'Admin.Advparameters.Feature'),
+                'filter_form_type' => TextType::class,
+            ],
+            [
+                'identifier' => 'error_code',
+                'name' => $this->trans('Error code', [], 'Admin.Advparameters.Feature'),
+                'filter_form_type' => TextType::class,
+            ],
+            [
+                'identifier' => 'date_add',
+                'name' => $this->trans('Date', [], 'Admin.Global'),
+                'filter_form_type' => TextType::class,
+            ],
+        ];
+
+        $columns = new ColumnCollection();
+        $position = 0;
+
+        foreach ($columnsArray as $columnArray) {
+            $columnArray['position'] = $position;
+
+            $column = Column::fromArray($columnArray);
+            $columns->add($column);
+
+            $position += 2;
```

Comment:
why do you need to make position of columns "even"

Reply:
probably we don't, i was playing with testing module and it's really easy to add new column with position 3, so it will be between column with position 2 and column with position 4.

however, i think better solution would be to create new method on `ColumnCollection` class something like addAfter($column, $identifier);

do you think it's better that way?

Reply:
yes :) I think that functions like addAfter or addBefore will be loved by PrestaShop developers :)